### PR TITLE
feat(invity): coinmarket input validity state improvement

### PR DESCRIPTION
**Pull request by LukasRada**
**User story**
> As a user, I want to be informed whether my input into any coin market form (buy, sell, exchange) is valid or not, so I can continue in trade or input valid values.

Invalid input fields are already rendered with a red border and error message. This commit adds rendering valid input fields with a green appearance.

Please be aware of change in `packages/components/src/components/form/Select/index.tsx` especially. I have been told that the `Select` component should be reviewed by someone who is responsible for the general components in Suite. The suggestion was @matejkriz. Please Matěj, could you review the Select component or recommend/assign someone else?

### CODE REVIEW

**Comment by szymonlesisz** [comment]
rebase this branch pls, there was and eslint update recently

File: packages/components/src/components/form/Select/index.tsx [hunk 1]
```diff
@@ -51,15 +52,15 @@ const selectStyle = (
             height,
             borderRadius: `${borderRadius}px`,
             borderWidth: `${borderWidth}px`,
-            borderColor: theme.STROKE_GREY,
+            borderColor: selectState ? getStateColor(selectState, theme) : theme.STROKE_GREY,
```
Comment (matejkriz) [inline packages/components/src/components/form/Select/index.tsx]: nitpick: would be nice to define `const borderColor = selectState ? getStateColor(selectState, theme) : theme.STROKE_GREY` above and use it here and for hover/focus state

File: packages/components/src/support/types/index.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ export type InputState = 'success' | 'warning' | 'error';
 
 export type InputVariant = 'small' | 'medium' | 'large';
 
+export type SelectState = InputState;
```
Comment (matejkriz) [inline packages/components/src/support/types/index.ts]: I don't see a benefit in defining SelectState type, InputState sounds fine for me even for select input, but nothing wrong of course.

File: packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/FiatInput/index.tsx [hunk 3]
```diff
@@ -25,25 +26,30 @@ const FiatInput = () => {
         trigger,
         updateSendCryptoValue,
         setValue,
+        getValues,
     } = useCoinmarketExchangeFormContext();
 
-    const error = errors.outputs && errors.outputs[0] ? errors.outputs[0].fiat : undefined;
+    const amountError = errors.outputs && errors.outputs[0] ? errors.outputs[0].amount : undefined;
+    const fiatError = errors.outputs && errors.outputs[0] ? errors.outputs[0].fiat : undefined;
```
Comment (matejkriz) [inline packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/FiatInput/index.tsx]: just an option how to write it more compact, definitely matter of taste so use it if you like it
```suggestion
    const amountError = errors.outputs?.[0]?.amount;
    const fiatError = errors.outputs?.[0]?.fiat;
```

or 
```suggestion
   const outputsErrors = errors.outputs?.[0] ?? {};
   const { amount: amountError, fiat: fiatError } = outputsErrors;
```

File: packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/FiatInput/index.tsx [hunk 4]
```diff
@@ -25,25 +26,30 @@ const FiatInput = () => {
...
-                if (error) {
+                if (fiatError) {
                     setValue(CRYPTO_INPUT, '');
                 } else {
                     updateSendCryptoValue(event.target.value, network.decimals);
                     clearErrors(FIAT_INPUT);
                 }
             }}
-            state={error ? 'error' : undefined}
+            state={getInputState(fiatError ?? amountError, fiat)}
```
Comment (matejkriz) [inline packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/FiatInput/index.tsx]: bit more precise for the case when `fiatError` is defined but falsy and `amountError` is truthy
```suggestion
            state={getInputState(fiatError || amountError, fiat)}
```

File: packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/ReceiveCryptoSelect/index.tsx [hunk 5]
```diff
@@ -130,16 +131,21 @@ const ReceiveCryptoSelect = () => {
         return false;
     };
 
-    const tokenAddress = getValues(CRYPTO_TOKEN);
-    const tokenData = account.tokens?.find(t => t.address === tokenAddress);
-
+    const { outputs, receiveCryptoSelect } = getValues();
+    const output = outputs && outputs[0];
```
Comment (matejkriz) [inline packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/ReceiveCryptoSelect/index.tsx]:
```suggestion
    const output = outputs?.[0];
```

**Review by matejkriz — COMMENTED** [review]
I have only couple of nitpicks, feel free to close them. Changes in Select component looks good to me and work as expected 👍

File: packages/components/src/components/form/Select/index.tsx (hunk 1 above)
Comment (LukasRada) [reply packages/components/src/components/form/Select/index.tsx]: Done.

File: packages/components/src/support/types/index.ts (hunk 2 above)
Comment (LukasRada) [reply packages/components/src/support/types/index.ts]: Removed. (Of course, another point of view is: select is also input in general)

File: packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/FiatInput/index.tsx (hunk 3 above)
Comment (LukasRada) [reply packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/FiatInput/index.tsx]: Agreed. Done.

File: packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/FiatInput/index.tsx (hunk 4 above)
Comment (LukasRada) [reply packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/FiatInput/index.tsx]: Done.

File: packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/ReceiveCryptoSelect/index.tsx (hunk 5 above)
Comment (LukasRada) [reply packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/ReceiveCryptoSelect/index.tsx]: Done at multiple places.

**Review by tomasklim — APPROVED** [review]
LGTM

**Comment by tomasklim** [comment]
@LukasRada Hello, there is some problem with e2e tests on your branch. 
Could you please check it? ```ERROR: for test-run  error parsing HTTP 404 response body: unexpected end of JSON input: “”```
Maybe rebasing branch to the newest develop will help.

**Comment by LukasRada** [comment]
Thank you for letting me know. Fixed.

**Comment by bosomt** [comment]
**QA OK**

im not sure if we have to render all fields like coin name in green but it's improvement 

obrazek

**Info:**
 - Suite version: web 21.9.0 (f9ecf819ac9cc809aa78295793c065c508f74e72)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:91.0) Gecko/20100101 Firefox/91.0
 - OS: MacIntel
 - Screen: 1680x1050
 - Device: model T 2.4.2 regular

**Pull request merged**